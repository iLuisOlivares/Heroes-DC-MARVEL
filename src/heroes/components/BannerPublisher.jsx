import React from 'react'

export const BannerPublisher = ({ img, publisher, color }) => {

    const style = {
        backgroundColor: color,
    };

    return (
        <div style={style} className='banner'>
            <div className='d-flex justify-content-evenly align-items-center h-100'>
                <div className='banner__img'>

                    <img src={img} alt="" />
                </div>

                <div className='banner__text'>
                    <p>{publisher} </p>
                    <div className="input-group">
                        <input type="search" className="form-control" placeholder="Buscar" aria-label="Search" aria-describedby="search-addon" />
                        <button type="button" className="btn btn-dark">Buscar</button>
                    </div>
                </div>


            </div>

        </div>


    )
}
