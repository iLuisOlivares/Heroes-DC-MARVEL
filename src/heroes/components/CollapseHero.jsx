
export const CollapseHero = ({ tittle, content, id }) => {


    return (
        <div className="accordion-item my-4">
            <h2 className="accordion-header">
                <button className="accordion-button" type="button" data-bs-toggle="collapse" data-bs-target={`#accordion${id}`} aria-expanded="true" aria-controls="panelsStayOpen-collapseOne">
                    {tittle}
                </button>
            </h2>
            <div id={`accordion${id}`} className="accordion-collapse collapse show" data-bs-parent="#accordionExample" >
                <div className="accordion-body">
                    {content}
                </div>
            </div>
        </div >

    )
}
