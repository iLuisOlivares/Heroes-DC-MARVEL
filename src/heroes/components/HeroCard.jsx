import { Link } from "react-router-dom"

export const HeroCard = ({
    id = 'dc-batman',
    superhero = 'Batman',
    description,
    image
}) => {


    description = description.length > 200 ? description.slice(0, 200) + "..." : description;

    return (

        <div className="col">
            <div className="card card_hero">
                <div className="row g-0 height-100">
                    <div className="col-md-8">
                        <div className="card-body">
                            <h5 className="card-title mt-4">{superhero}</h5>
                            <p className="card-text card_text">{description}</p>
                            <Link style={{ "color": "#8C30F5", "textDecoration": "none" }} className="text" to={`/hero/${id}`}>Explorar âž¤</Link>
                        </div>
                    </div>
                    <div className="col-md-4">
                        <img src={image} className="img-fluid rounded card-image" alt="..." />

                    </div>
                </div>
            </div>
        </div>

    )
}
