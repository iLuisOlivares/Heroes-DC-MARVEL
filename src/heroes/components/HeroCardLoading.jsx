import React from 'react'
import { Link } from 'react-router-dom'

export const HeroCardLoading = () => {
    return (
        <div className="col">
            <div className="card card_hero">
                <div className="row g-0 height-100">
                    <div className="col-md-8">
                        <div className="card-body">
                            <h5 class="card-title placeholder-glow mt-4">
                                <span class="placeholder col-6"></span>
                            </h5>
                            <p class="card-text placeholder-glow">
                                <span class="placeholder col-10"></span>
                                <span class="placeholder col-10"></span>
                                <span class="placeholder col-10"></span>
                                <span class="placeholder col-10"></span>
                                <span class="placeholder col-10"></span>
                                <span class="placeholder col-10"></span>
                            </p>
                            <Link style={{ "color": "#8C30F5", "textDecoration": "none" }} className="text disabled placeholder" >Explorar âž¤</Link>
                        </div>
                    </div>
                    <div className="col-md-4">
                        <img src={'https://www.solidbackgrounds.com/images/1920x1080/1920x1080-gray-solid-color-background.jpg'} className="img-fluid rounded card-image" alt="..." />

                    </div>
                </div>
            </div>
        </div>
    )
}


