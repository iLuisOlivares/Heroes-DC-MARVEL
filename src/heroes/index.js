
// export * from './pages/Dc';
// export * from './pages/Marvel';

export * from './pages';