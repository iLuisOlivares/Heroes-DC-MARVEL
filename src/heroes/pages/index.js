
export * from './Dc';
export * from './Marvel';
export * from './SearchPage';
export * from './HeroPage';