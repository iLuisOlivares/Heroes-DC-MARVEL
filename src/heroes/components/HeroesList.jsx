import React, { useMemo } from 'react'
import { getHeroesByPublisher } from '../helpers/getHeroes'
import { HeroCard } from './HeroCard';
import useFetch from '../../UI/hooks/useFetch';
export const HeroesList = ({ publisher }) => {

    // const heroes = useMemo(() => getHeroesByPublisher(publisher), [publisher]);
    const { data, error, isLoading } = useFetch('http://127.0.0.1:8000/api/characters/?search=' + publisher.replace(' ', '+'));
    return (
        <div style={{ "backgroundColor": "#0B1E51", "padding": "20px", "minHeight": "100vh" }}>
            {isLoading ? <p className='text-light'>Loading...</p>
                : error
                    ? <p>Error: {error.message} </p>
                    : <div class="row row-cols-1 row-cols-md-3 g-4">
                        {data.map(hero => (
                            <HeroCard publisher={hero.publisher} key={hero.superhero_name} id={hero.superhero_name} superhero={hero.superhero_name} alter_ego={hero.alias} description={hero.description} image={hero.image}></HeroCard>
                        ))
                        }
                    </div>}
        </div>

    )
};

