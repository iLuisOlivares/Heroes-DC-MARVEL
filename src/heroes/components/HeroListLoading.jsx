import React from 'react'
import { HeroCardLoading } from './HeroCardLoading'

export const HeroListLoading = () => {


    return (
        <div class="row row-cols-1 row-cols-md-3 g-4">
            <HeroCardLoading></HeroCardLoading>
            <HeroCardLoading></HeroCardLoading>
            <HeroCardLoading></HeroCardLoading>
            <HeroCardLoading></HeroCardLoading>
            <HeroCardLoading></HeroCardLoading>
            <HeroCardLoading></HeroCardLoading>
        </div>

    )
}
