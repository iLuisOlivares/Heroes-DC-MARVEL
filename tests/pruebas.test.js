import { render, screen, waitFor } from '@testing-library/react';
import { HeroCard } from '../src/heroes/components/HeroCard';
import { MemoryRouter } from 'react-router-dom';
import { Dc, Marvel } from '../src/heroes';


// TEST DE PRUEBA A HeroList
describe('Pruebas en publisher pages', () => {


    test('Debe mostrar la seccion respectiva de Marvel', async () => {
        const publisher = 'Marvel Comics';
        const { container, getByText } = render(
            <MemoryRouter>
                <Marvel></Marvel>
            </MemoryRouter>

        );

        expect(getByText(publisher)).toBeTruthy();
    });

    test('Debe mostrar la seccion respectiva de Dc', async () => {
        const publisher = 'Dc Comics';
        const { container, getByText } = render(
            <MemoryRouter>
                <Dc></Dc>
            </MemoryRouter>

        );

        expect(getByText(publisher)).toBeTruthy();
    });


});


// TEST DE PRUEBA A HeroCard
describe('Pruebas en <HeroCard>', () => {

    const description = "Batman es un superhéroe que lucha contra el crimen en la ciudad de Gotham. Es conocido por su astucia, su habilidad para el combate cuerpo a cuerpo y sus gadgets tecnológicos."

    test('debe Hacer match con el snapshot', () => {

        const { container } = render(<MemoryRouter>
            <HeroCard publisher={"Dc Comics"} key={"Batman"} id={"Batman"} superhero={"Batman"} alter_ego={"Bruce Wayne"} description={description} image={"https://res.cloudinary.com/iluiss/image/upload/v1687543445/Heroes/unblscke0mqpi7vqoxtd.jpg"} />
        </MemoryRouter>)

        expect(container).toMatchSnapshot();
    });

    test('La descripcion debe estar presente en el componente', () => {

        const { container, getByText } = render(<MemoryRouter>
            <HeroCard publisher={"Dc Comics"} key={"Batman"} id={"Batman"} superhero={"Batman"} alter_ego={"Bruce Wayne"} description={description} image={"https://res.cloudinary.com/iluiss/image/upload/v1687543445/Heroes/unblscke0mqpi7vqoxtd.jpg"} />
        </MemoryRouter>)

        expect(getByText(description)).toBeTruthy();
    });



});



