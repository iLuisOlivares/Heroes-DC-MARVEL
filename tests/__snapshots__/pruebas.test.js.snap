// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroCard> debe Hacer match con el snapshot 1`] = `
<div>
  <div
    class="col"
  >
    <div
      class="card card_hero"
    >
      <div
        class="row g-0 height-100"
      >
        <div
          class="col-md-8"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title mt-4"
            >
              Batman
            </h5>
            <p
              class="card-text card_text"
            >
              Batman es un superhéroe que lucha contra el crimen en la ciudad de Gotham. Es conocido por su astucia, su habilidad para el combate cuerpo a cuerpo y sus gadgets tecnológicos.
            </p>
            <a
              class="text"
              href="/hero/Batman"
              style="color: rgb(140, 48, 245); text-decoration: none;"
            >
              Explorar ➤
            </a>
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <img
            alt="..."
            class="img-fluid rounded card-image"
            src="https://res.cloudinary.com/iluiss/image/upload/v1687543445/Heroes/unblscke0mqpi7vqoxtd.jpg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
